### ISSUE TITLE
[FEA] libcudf function for copying to a "slice" of a column

### ISSUE THREAD

**Issue by shwina** [rapidsai/cudf#1934]
It would be useful to have a libcudf function to copy values to a subset of a column. The use cases in cuDF would be:

```
a = cudf.Series([1, 2, 3, 4])
b = cudf.Series([5 ,6, 7, 8])

a[2:4] = b[1:3]

# a is now [1, 2, 6, 7]
```

```
a = cudf.Series([1, 2, 3, 4])
b = cudf.Series([5 ,6])

a[2:4] = b

# a is now [1, 2, 5, 6]
```

**Comment by kkraus14** [comment]
Wouldn't `b[1:3]` just return a dense buffer via a slice or gather call that could be passed into the scatter call without issue? I don't see the need for another API here but maybe I'm missing something.

**Comment by shwina** [comment]
In an offline conversation @harrism explained that it would be inefficient since scatter would be doing `out[map[i]] = in[i]` where we could just be doing `out[i] = in[i]`

**Comment by kkraus14** [comment]
> In an offline conversation @harrism explained that it would be inefficient since scatter would be doing `out[map[i]] = in[i]` where we could just be doing `out[i] = in[i]`

Would a more generalized `scatter_slice` type operation work that the object passed to the `__setitem__` call would be any dense buffer of the same size of the slice?

**Comment by shwina** [comment]
Yup - _that_ is what I'm looking for. Have re-worded the issue.

**Comment by kkraus14** [comment]
@harrism this would look similar to #1353 but for "scattering" instead of "gathering"

**Comment by harrism** [comment]
Sure. See #1908.  Note that this is not really a scatter, as @shwina points out.  This is assignment to a contiguous range. Scatter can do that, but the indirection would be inefficient when you know the range is contiguous.

**Comment by harrism** [comment]
C++ API will be something like `void copy_range(gdf_column *out_column, gdf_column const &in_column,  gdf_index_type out_begin, gdf_index_type out_end, gdf_index_type in_begin);

**Comment by harrism** [comment]
Closed by #1908
